Automattic/wp-calypso
# Framework: Refactor away from _.isArray()

#### Changes proposed in this Pull Request

Lodash's `isArray` is essentially fully natively supported by `Array.isArray`. This PR replaces the usage, and adds an ESLint rule to warn against using `isArray` from lodash. It might look like a big PR, but replacing one with the other is pretty harmless (neither will throw with any value), so no extensive testing is actually necessary.

If you want to find out why we're moving away from Lodash, see https://github.com/Automattic/wp-calypso/pull/50368 and p4TIVU-9Bf-p2.

#### Testing instructions

* Try importing `isArray` from `lodash` and verify you're getting an ESLint error.
* Verify all tests pass.

#### Notes

There is one ESLint error in one of the files we're touching, and we're intentionally leaving it be. That needs to be refactored away from `create-react-class` (`react/prefer-es6-class`).
